pkg/docker/docker.go

```diff
@@ -581,20 +518,56 @@ func (d *stiDocker) CheckImage(name string) (*api.Image, error) {
 	return nil, nil
 }
 
+func base64EncodeAuth(auth dockertypes.AuthConfig) (string, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(auth); err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(buf.Bytes()), nil
+}
+
 // PullImage pulls an image into the local registry
 func (d *stiDocker) PullImage(name string) (*api.Image, error) {
 	name = getImageName(name)
-	err := d.kubeDockerClient.PullImage(name, d.pullAuth, dockertypes.ImagePullOptions{})
+
+	// RegistryAuth is the base64 encoded credentials for the registry
+	base64Auth, err := base64EncodeAuth(d.pullAuth)
 	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	resp, err := d.InspectImage(name)
+	opts := dockertypes.ImagePullOptions{RegistryAuth: base64Auth}
+	ctx := context.Background() // no timeout
```

Comment:
i think this should be the 2 minute value like we did on inspect image (which essentially can do a pull image under the covers) vs. no timeout